rust/protocol/src/protocol.rs

```diff
@@ -4,19 +4,63 @@
 //
 
 use crate::proto;
-use crate::{IdentityKey, PrivateKey, PublicKey, Result, SignalProtocolError};
+use crate::state::{PreKeyId, SignedPreKeyId};
+use crate::{DeviceId, IdentityKey, PrivateKey, PublicKey, Result, SignalProtocolError};
 
-use std::convert::TryFrom;
+use std::convert::{TryFrom, TryInto};
 
 use hmac::{Hmac, Mac, NewMac};
+use num_enum::TryFromPrimitiveError;
 use prost::Message;
 use rand::{CryptoRng, Rng};
 use sha2::Sha256;
 use subtle::ConstantTimeEq;
 use uuid::Uuid;
 
-pub const CIPHERTEXT_MESSAGE_CURRENT_VERSION: u8 = 3;
-pub const SENDERKEY_MESSAGE_CURRENT_VERSION: u8 = 3;
+pub const CIPHERTEXT_MESSAGE_CURRENT_VERSION: MessageVersion = MessageVersion::V3;
+pub const SENDERKEY_MESSAGE_CURRENT_VERSION: MessageVersion = MessageVersion::V3;
```

Comment:
These should (sadly) not be the same type, as they already don't represent the same kind of message or session (even if they share an HKDF). Honestly given that message version 2 isn't supported I'm not sure what to think about `HKDF::new`.

Reply:
I have reverted this change and isolated it into https://github.com/cosmicexplorer/libsignal-client/commit/a29e6d3b0ed1f4dd5ce5c4944b90accab8c3b1a9 which I will push in a followup PR!